### ISSUE TITLE
Unable to create folders with grafana 8.5

### ISSUE THREAD

**Issue by dschaaff** [grafana/terraform-provider-grafana#463]
### Terraform Version
* Terraform: 1.1.7
* Terraform Grafana Provider: 1.21.1
* Grafana: 8.5.0

### Affected Resource(s)

* grafana_dashboard
* grafana_folder

```
error: Provider produced inconsistent result after apply
│ 
│ When applying changes to grafana_folder.envoy, provider "provider[\"registry.terraform.io/grafana/grafana\"]" produced an unexpected new value: Root resource was present, but now absent.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵```

### Terraform Configuration Files
```hcl
resource "grafana_folder" "envoy" {
  title = "Envoy"
}
```

**Comment by cthulhu** [comment]
Same here, 

- Terraform version Terraform v1.1.9
- Provider version: v1.21.1
- Grafana Cloud

Affected resources:

- grafana_folder

> Error: Provider produced inconsistent result after apply
> │
> │ When applying changes to module.grafana_monitoring.grafana_folder.team_folder, provider "provider[\"registry.terraform.io/grafana/grafana\"]" produced an unexpected new value: Root resource was present, but now absent.
> │
> │ This is a bug in the provider, which should be reported in the provider's own issue tracker.
> ╵

**Comment by mju-spyrosoft** [comment]
Same here
```
$ terraform -version
Terraform v1.1.8
on windows_amd64
+ provider registry.terraform.io/grafana/grafana v1.21.1
+ provider registry.terraform.io/hashicorp/aws v4.11.0
```
edit: Cloud Grafana version 8.5.0

Result of apply command
```
module.toolchain.module.grafana.grafana_folder.kkrdash_main: Creating...
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to module.toolchain.module.grafana.grafana_folder.kkrdash_main, provider "module.toolchain.module.grafana.provider[\"registry.terraform.io/grafana/grafana\"]" produced an unexpected new value: Root resource
│ was present, but now absent.
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
Releasing state lock. This may take a few moments...

```

btw I noticed that folder does actually get created inside of grafana if that matters in any way. After that i tried importing as a workaround but i think im doing something wrong 
```
$terraform import module.toolchain.module.grafana.grafana_folder.kkrdash_main 160
module.toolchain.module.grafana.grafana_folder.kkrdash_main: Refreshing state... [id=160]
╷
│ Error: Cannot import non-existent remote object
│
│ While attempting to import an existing object to "module.toolchain.module.grafana.grafana_folder.kkrdash_main", the provider detected that no object exists with the given id. Only pre-existing objects can be imported; check that
│ the id is correct and that it is associated with the provider's configured region or endpoint, or use "terraform apply" to create a new remote object for this resource.
```

**Comment by julienduchesne** [comment]
I'm investigating this

**Comment by julienduchesne** [comment]
This is a bug in Grafana 8.5: https://github.com/grafana/grafana/pull/48124, it probably wasn't spotted because the folder by ID endpoint is a lesser used one. In the meantime, I have a workaround I'll be pushing as a provider patch

**Comment by julienduchesne** [comment]
This is released in v1.21.2 (and v1.22.0 which is the Grafana OnCall release)
